docs/_extension/list_all_units.py

```diff
@@ -0,0 +1,40 @@
+from barril.units import Scalar
+from docutils import nodes
+from docutils.statemachine import ViewList
+from sphinx.util import nested_parse_with_titles
+from sphinx.util.docutils import SphinxDirective
+
+
+class ListAllUnits(SphinxDirective):
+    def run(self):
+        rst = ViewList()
+        unit_database = Scalar(10, "m").GetUnitDatabase()
+        source = "fakefile.rst"
+
+        # Create the rst content
+        for categories in sorted(unit_database.IterCategories(), key=str.casefold):
+            title = unit_database.GetCategoryInfo(categories).caption
+            rst.append(f".. rubric:: {title}", source)
+            rst.append(f"", source)
+
+            for unit in unit_database.GetCategoryInfo(categories).valid_units_set:
+                unit = unit_database.unit_to_unit_info[unit].unit
+                name = unit_database.unit_to_unit_info[unit].name
```

Comment:
* This should be sorted by `unit`
* Why `unit = unit_database.unit_to_unit_info[unit].unit`?

Reply:
> This should be sorted by unit
Good call =)

>> Why `unit = unit_database.unit_to_unit_info[unit].unit?`
🤦‍♂

Reply:
Commit 91515c56de1862d4efc11b7dc086f11664c45c93 fix this issue